tests/Bridge/Laravel/Console/Commands/SearchIndexFillCommandTest.php

```diff
@@ -0,0 +1,118 @@
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Tests\LoyaltyCorp\Search\Stubs\Handlers\HandlerStub;
+use Tests\LoyaltyCorp\Search\Stubs\Handlers\OtherHandlerStub;
+use Tests\LoyaltyCorp\Search\Stubs\IndexerStub;
+use Tests\LoyaltyCorp\Search\TestCase;
+
+/**
+ * @covers \LoyaltyCorp\Search\Bridge\Laravel\Console\Commands\SearchIndexCommand
+ * @covers \LoyaltyCorp\Search\Bridge\Laravel\Console\Commands\SearchIndexFillCommand
+ *
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects) Required for thorough testing
+ */
+class SearchIndexFillCommandTest extends TestCase
+{
+    /**
+     * Ensure
+     *
+     * @return void
+     *
+     * @throws \ReflectionException
+     */
+    public function testIndexerPopulateCalled(): void
+    {
+        $indexer = new IndexerStub();
+        $output = new NullOutput();
+        $container = new Container();
+        $container->tag([
+            HandlerStub::class,
+            OtherHandlerStub::class
+        ], ['search_handler']);
+        $command = $this->createInstance([], $output, $indexer, $container);
+
+        $command->handle();
+
+        // No indices because the tagged search handler does not have the HandlerInterface
+        self::assertSame(2, $indexer->getPopulatedCount());
+    }
+
+    /**
+     * Ensure a tagged search handler that does not implement the interface is not actually attempted execution on
+     *
+     * @return void
+     *
+     * @throws \ReflectionException
+     */
+    public function testNonSearchHandlerRegisteredIsSkipped(): void
+    {
+        $indexer = new IndexerStub();
+        $output = new NullOutput();
+        $container = new Container();
+        $container->tag([
+            IndexerStub::class
+        ], ['search_handler']);
+        $command = $this->createInstance([], $output, $indexer, $container);
+
+        $command->handle();
+
+        // No indices because the tagged search handler does not have the HandlerInterface
+        self::assertSame(0, $indexer->getCreatedCount());
+    }
+
+    /**
+     * Create command instance
+     *
+     * @param mixed[] $options Options to pass to the command
+     * @param \Symfony\Component\Console\Output\OutputInterface $output The interface to output the result to
+     * @param \LoyaltyCorp\Search\Interfaces\IndexerInterface|null $indexer
+     * @param \Illuminate\Contracts\Container\Container|null $container
+     *
+     * @return \LoyaltyCorp\Search\Bridge\Laravel\Console\Commands\SearchIndexFillCommand
+     *
+     * @throws \ReflectionException If class being reflected does not exist
+     */
+    private function createInstance(
+        array $options,
+        OutputInterface $output,
+        ?IndexerInterface $indexer = null,
+        ?ContainerInterface $container = null
```

Comment:
If there are no methods that use the nullable path for these properties they should just be not nullable

Reply:
Can this method be generalised into a base test case?

Reply:
It could, but we'd end up with a bit of conditional checking. due to optional console arguments, input/output (when required), and the input/output class type checking itself and the constructing of console application object itself - I'm unsure if it'd be worth it.
I'll clean up these methods to only have the exact code required, and hopefully that will simplify things